subsys/settings/src/settings_store.c

```diff
@@ -50,26 +56,34 @@ int settings_load(void)
 	 *    apply config
 	 *    commit all
 	 */
-
+	k_mutex_lock(&settings_lock, K_FOREVER);
 	SYS_SLIST_FOR_EACH_CONTAINER(&settings_load_srcs, cs, cs_next) {
-		cs->cs_itf->csi_load(cs);
+		cs->cs_itf->csi_load(cs, subtree);
 	}
-	return settings_commit();
+	k_mutex_unlock(&settings_lock);
+	return settings_commit_subtree(subtree);
```

Comment:
This is not safe - committing outside of mutex - what if another thread tries to load the same subtree in the midtime?

Reply:
Ok, no problem to put the commit inside the lock.